include/zephyr/arch/arm64/hypercall.h

```diff
@@ -14,6 +14,9 @@ int HYPERVISOR_event_channel_op(int op, void *param);
 int HYPERVISOR_hvm_op(int op, void *param);
 int HYPERVISOR_memory_op(int op, void *param);
 int HYPERVISOR_grant_table_op(int op, void *uop, unsigned int count);
+#ifdef CONFIG_XEN_INITIAL_DOMAIN
```

Comment:
This is not needed. Xen has own access system. Moreover, it is possible to configure XSM to allow other guests to issue `domctl` commands.

`XENVER_` calls are available or all domains by default.

Reply:
@lorc agree with xen_version, but domctl and sysctl are unclear. I'm not sure about hypercall, but all wrapper functions for sysctl and domctl have a build check if they are built in control domain.

Reply:
@lorc @firscity Should I leave this or remove?